libs/blocks/table-of-contents/table-of-contents.js

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/*
+ * Table of Contents
+ */
+import { createTag } from '../../utils/utils.js';
+
+function findAnchorTarget(text) {
+  let linkText = text.toLowerCase();
+  linkText = linkText.charAt(0) === '_' ? linkText.substring(1) : linkText;
+  linkText = linkText.replaceAll('_', '-');
+  linkText = linkText.replaceAll(/[ /|&;$%@"<>()+,.]/g, '');
+  return document.querySelector(`[id^="${linkText}"]`);
+}
+
+const DOWN_ARROW_ICON = `<svg xmlns="http://www.w3.org/2000/svg" width="10.875" height="13.323" viewBox="0 0 10.875 13.323">
+  <g id="Group_177418" data-name="Group 177418" transform="translate(223.353 -1276.435) rotate(90)">
+    <line id="line_color_border_" data-name="line [color_border]" x2="10.909" transform="translate(1277.435 217.875)" fill="none" stroke="#1473e6" stroke-linecap="round" stroke-width="2"/>
+    <g id="arrow_head" data-name="arrow head" transform="translate(1284.32 213.892) rotate(45)">
+      <line id="line_color_border_2" data-name="line [color_border]" x2="5.69" transform="translate(0 0)" fill="none" stroke="#1473e6" stroke-linecap="round" stroke-width="2"/>
+      <line id="line_color_border_3" data-name="line [color_border]" y2="5.69" transform="translate(5.69 0)" fill="none" stroke="#1473e6" stroke-linecap="round" stroke-width="2"/>
+    </g>
+  </g>
+</svg>`;
+
+function tocItem(title, subtitle, target) {
+  const onClick = () => {
+    const pageTop = document.querySelector('header')?.offsetHeight ?? 0;
+    const targetPosition = target.getBoundingClientRect().top;
+    const offsetPosition = targetPosition + window.pageYOffset - pageTop;
+
+    window.scrollTo(0, offsetPosition);
+    target?.setAttribute('tabindex', -1);
+    target?.focus();
+  };
+
+  const sectionTitle = `<p class="section-title">${title}</p>`;
+  const sectionDescription = subtitle ? `<p class="section-description">${subtitle}</p>` : '';
+  const html = `<div class="toc-link-text">
+    ${sectionTitle}
+    ${sectionDescription}
+  </div>
+  <div class="toc-arrow">
+    ${DOWN_ARROW_ICON}
+  </div>`;
+
+  const item = createTag('section', { class: 'toc-item', tabindex: '0' }, html);
```

Comment:
Can we just use an <a> tag? This will make it easier to pick up our analytics.js dom tagging.

Reply:
Along these same lines but related to accessibility. I can tab focus to these but can't navigate to the related section with out the mouse click.

Reply:
In the example on AEM they hide the `<a>` tag at the bottom of the section, does that work for analytics and accessibility or does it need to be apart of the section?
![image](https://user-images.githubusercontent.com/681036/193091047-34259815-5193-4cf7-b342-b6fd92cb84d2.png)

Reply:
@Brandon32 

I think the approach outlined in the attached link might be the best approach for this, and solves several gotcha's along the way. 
https://css-tricks.com/block-links-the-search-for-a-perfect-solution/#method-4-sprinkle-javascript-on-the-second-method

cc: @ryanmparrish @auniverseaway

Reply:
@rgclayton Sounds good, I talked to accessibility and had to do a bit of refactoring to add a nav and list. 
I have a preventDefault to offset the scroll, that wont cause any gotcha, right?

Reply:
No, I don't think so. 

Only things I see now are the cursor doesn't change to pointer when hovering and if I select the text and release my mouse it activates the click to jump, which is unexpected. Check out the [codepen](https://codepen.io/temperedpixel/pen/XWqEzEb?editors=1111) in that article. They have a pretty simple and slick solution for that.

Reply:
Are you talking about `cursor: pointer;`? I think I forgot to re-add that style after trying to make the whole thing a link. 
I have to offset the scroll and I want to make sure there isn't any gotcha's with analytics or accessibility.

Reply:
Yes, `cursor: pointer`. 

I think selecting text is a problem. Currently, selecting text inside of the link content causes the click listener to fire and jump to its named anchor. You can use this `window.getSelection().toString()` to check if text is selected and not fire the click listener. See the codepen/article i mentioned in my previous comments.

Reply:
Oh, I thought that was related to having multiple links on the card.
FYI this is the example I was given by our accessibility contact.
https://heydon.github.io/Inclusive-Components/cards-redundant-click/
~~~
    card.onmousedown = () => down = +new Date();
    card.onmouseup = () => {
        up = +new Date();
        if ((up - down) < 200) {
            link.click();
        }
    }
~~~